lightning-background-processor/src/lib.rs

```diff
@@ -352,8 +347,7 @@ macro_rules! define_run_body {
 			// falling back to our usual hourly prunes. This avoids short-lived clients never
 			// pruning their network graph. We run once 60 seconds after startup before
 			// continuing our normal cadence.
-			let prune_timer = if have_pruned { NETWORK_PRUNE_TIMER } else { FIRST_NETWORK_PRUNE_TIMER };
-			if $timer_elapsed(&mut last_prune_call, prune_timer) {
+			if !have_pruned || $timer_elapsed(&mut last_prune_call, NETWORK_PRUNE_TIMER) {
```

Comment:
I couldn't find a great way to expose when an "initial sync" would be considered over for P2P, since IIUC we basically do an "unofficial" sync of just getting either a full dump (no-std) or latest 2 weeks (std) through the gossip timestamp filter for our first 5 peers we connect to? For now I just left the 60s initial prune timer for P2P but allowed pruning before this once RGS initial sync has completed.